datafusion/physical-expr/src/expressions/binary/kernels.rs

```diff
@@ -27,8 +27,34 @@ use arrow::datatypes::DataType;
 use datafusion_common::internal_err;
 use datafusion_common::{Result, ScalarValue};
 
+use arrow_schema::ArrowError;
 use std::sync::Arc;
 
+/// Returns the elementwise concatenation of a [`StringViewArary`].
+///
+/// An index of the resulting [`StringViewArray`] is null if any of
+/// `StringViewArary` are null at that location.
+///
+/// ```text
+/// e.g:
+///
+///   ["Hello"] + ["World"] = ["HelloWorld"]
+///
+///   ["a", "b"] + [None, "c"] = [None, "bc"]
+/// ```
+///
+/// An error will be returned if `left` and `right` have different lengths
+///
+/// TODO: port this upstream to arrow-rs (TODO FILE TICKET AND REFERENCE HERE)
```

Comment:
The idea would be to fill out this function (and eventually port it upstream to arrow-rs)